### PR TITLE
added user for keygen along the flow

### PR DESCRIPTION
**Pull request by darkCavalier11**
when the server search for files `key` and `key.pub` if file cant located it asks the user if they want to generate keys and the server handles the keygen process and asks for restarting the server.

### CODE REVIEW

**Comment by birm** [comment]
Since the server is usually running without an interactive terminal, what if we use an environment variable such as `GENERATE_KEYS_IF_MISSING`?

**Comment by darkCavalier11** [comment]
good idea. I will work on this and update soon.

File: handlers/authHandlers.js [hunk 1]
```diff
@@ -13,6 +16,17 @@ var EXPIRY = process.env.EXPIRY || '1d';
 var DEFAULT_USER_TYPE = process.env.DEFAULT_USER_TYPE || 'Null';
 var PUBKEY;
 var PRIKEY;
+var GENERATE_KEY_IF_MISSING = process.env.GENERATE_KEY_IF_MISSING || true;
+
+if (!fs.existsSync('./keys/key.pub') && !fs.existsSync('./keys/key.pub') && GENERATE_KEY_IF_MISSING) {
```
Comment (birm) [inline handlers/authHandlers.js:21]:
```suggestion
if (!fs.existsSync('./keys/key') && !fs.existsSync('./keys/key.pub') && GENERATE_KEY_IF_MISSING) {
```

File: .eslintrc.js [hunk 2]
```diff
@@ -17,9 +17,10 @@ module.exports = {
   'rules': {
     "require-jsdoc" : 0,
     "valid-jsdoc" : 0,
-    "max-len" : ["error", { "code": 120 ,"ignoreTemplateLiterals": true},],
+    "max-len" : ["error", { "code": 178 ,"ignoreTemplateLiterals": true},],
```
Comment (birm) [inline .eslintrc.js:20]: Is there a way to assemble the command while following the max len rule?

Comment (darkCavalier11) [reply .eslintrc.js]: i did the suggested changes below

File: handlers/authHandlers.js [hunk 3]
```diff
@@ -13,6 +16,17 @@ var EXPIRY = process.env.EXPIRY || '1d';
 var DEFAULT_USER_TYPE = process.env.DEFAULT_USER_TYPE || 'Null';
 var PUBKEY;
 var PRIKEY;
+var GENERATE_KEY_IF_MISSING = process.env.GENERATE_KEY_IF_MISSING || true;
+
+if (!fs.existsSync('./keys/key') && !fs.existsSync('./keys/key.pub') && GENERATE_KEY_IF_MISSING) {
+  try {
+    exec(`${preCommand}'/CN=www.camicroscope.com/O=caMicroscope Local Instance Key./C=US'${postCommand}`);
+    console.log('keys are generated. please restart the server.' + '\n');
+    process.exit(0);
```
Comment (birm) [inline handlers/authHandlers.js:25]: What happens if the server doesn't exit here? Is there a benefit to it exiting?

Comment (darkCavalier11) [reply handlers/authHandlers.js:25]: all the validation is handled by the server. if the server doesn't exist  `authHandler.js` never run so no required for key validation. correct me if I am wrong.

Comment (birm) [reply handlers/authHandlers.js:25]: I mean, you create the key, so the next steps would be able to find key and key.pub if the process kept running, right?
I think you may need to use execSync though.

Comment (darkCavalier11) [reply handlers/authHandlers.js:25]: sure. i made the changes below.

**Comment by birm** [comment]
Does it now work if it doesn't exit after key creation?

**Comment by darkCavalier11** [comment]
as the command of keygen execute synchronously no need to exit and restart the server again. here in this commit, I remove the `process.exit(0)`.  it will execute along with the flow of execution as needed.

**Pull request merged**